src/Sentry/MetricHelper.cs

```diff
@@ -44,28 +44,47 @@ internal static long GetTimeBucketKey(this DateTimeOffset timestamp)
 
 #if NET7_0_OR_GREATER
-    [GeneratedRegex(InvalidKeyCharactersPattern, RegexOptions.Compiled)]
-    private static partial Regex InvalidKeyCharacters();
-    internal static string SanitizeKey(string input) => InvalidKeyCharacters().Replace(input, "_");
+    [GeneratedRegex(InvalidMetricKeyOrNameCharactersPattern, RegexOptions.Compiled)]
+    private static partial Regex InvalidMetricKeyOrNameCharacters();
+    internal static string SanitizeMetricKeyOrName(string input) => InvalidMetricKeyOrNameCharacters().Replace(input, "_");
 
-    [GeneratedRegex(InvalidValueCharactersPattern, RegexOptions.Compiled)]
-    private static partial Regex InvalidValueCharacters();
-    internal static string SanitizeValue(string input) => InvalidValueCharacters().Replace(input, "");
+    [GeneratedRegex(InvalidTagKeyCharactersPattern, RegexOptions.Compiled)]
+    private static partial Regex InvalidTagKeyCharacters();
+    internal static string SanitizeTagKey(string input) => InvalidTagKeyCharacters().Replace(input, "");
 
     [GeneratedRegex(InvalidMetricUnitCharactersPattern, RegexOptions.Compiled)]
     private static partial Regex InvalidMetricUnitCharacters();
-    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters().Replace(input, "_");
+    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters().Replace(input, "");
 #else
-    private static readonly Regex InvalidKeyCharacters = new(InvalidKeyCharactersPattern, RegexOptions.Compiled);
-    internal static string SanitizeKey(string input) => InvalidKeyCharacters.Replace(input, "_");
+    private static readonly Regex InvalidMetricKeyOrNameCharacters = new(InvalidMetricKeyOrNameCharactersPattern, RegexOptions.Compiled);
+    internal static string SanitizeMetricKeyOrName(string input) => InvalidMetricKeyOrNameCharacters.Replace(input, "_");
 
-    private static readonly Regex InvalidValueCharacters = new(InvalidValueCharactersPattern, RegexOptions.Compiled);
-    internal static string SanitizeValue(string input) => InvalidValueCharacters.Replace(input, "");
+    private static readonly Regex InvalidTagKeyCharacters = new(InvalidTagKeyCharactersPattern, RegexOptions.Compiled);
+    internal static string SanitizeTagKey(string input) => InvalidTagKeyCharacters.Replace(input, "");
 
     private static readonly Regex InvalidMetricUnitCharacters = new(InvalidMetricUnitCharactersPattern, RegexOptions.Compiled);
-    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters.Replace(input, "_");
+    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters.Replace(input, "");
 #endif
 
+    private static readonly Lazy<KeyValuePair<string, string>[]> LazyTagValueReplacements = new(() =>
+    [
+        new KeyValuePair<string, string>("\n", "<LF>"),
+        new KeyValuePair<string, string>("\r", "<CR>"),
+        new KeyValuePair<string, string>("\t", "<HT>"),
```

Comment:
Are those the right way around?

Reply:
I think so. [The docs](https://develop.sentry.dev/sdk/metrics/#normalization) say `Reserved characters for Tag Values are \n, \r, \t, \, |, ,. `

It's a bit confusing because directly underneath that statement is a table where sometimes things from that list are on the left hand side of the table and other times they're on the right hand side of the table. I believe this is a problem with the docs so I opened this PR to correct it:
- https://github.com/getsentry/develop/pull/1218

Reply:
The implementation is not correct. If there is a tab in the tag value, we want a string of `\t` to end up on the tag value. Not `<HT>`.

Reply:
Yes, these three.

Reply:
Since this PR is already merged, I've addressed in a separate PR:
- https://github.com/getsentry/sentry-dotnet/pull/3281